From 76581e742b22a2ca1b3b5ff5017ff252356dae45 Mon Sep 17 00:00:00 2001
From: Neeraj Solanki <neeraj.solanki@intel.com>
Date: Fri, 7 Jul 2017 13:25:04 +0530
Subject: [PATCH 3/4] Considreation for VENDOR_ART_PATH (Allow vendor to deploy
 another ART VM with proprietary optimizations)

Change-Id: I219e31cb044b58b98ad4ce019ab22b5006167753
Tracked-On:https://jira01.devtools.intel.com/browse/OAM-48711
Category: device enablement
Domain: AOSP.ART-Runtime
Origin: internal
Upstream-Candidate: no, proprietary
Signed-off-by: Neeraj Solanki <neeraj.solanki@intel.com>
---
 Android.mk | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/Android.mk b/Android.mk
index 8735d7c..202b826 100644
--- a/Android.mk
+++ b/Android.mk
@@ -18,6 +18,15 @@ LOCAL_PATH := $(call my-dir)
 
 art_path := $(LOCAL_PATH)
 
+# Allow vendor to deploy another ART VM with proprietary optimizations.
+ifneq ($(VENDOR_ART_PATH),)
+  # Set this way to reduce conflicts in file.
+  art_path := $(VENDOR_ART_PATH)
+else
+  # If vendor is not deploying separate VM, just use ART.
+  VENDOR_ART_PATH := $(art_path)
+endif
+
 ########################################################################
 # clean-oat rules
 #
-- 
1.9.1

